src/Microsoft.Extensions.Http/Logging/HttpHeadersLogValue.cs

```diff
@@ -0,0 +1,125 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Text;
+
+namespace Microsoft.Extensions.Http.Logging
+{
+    internal class HttpHeadersLogValue : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private readonly Kind _kind;
+
+        private string _formatted;
+        private List<KeyValuePair<string, object>> _values;
+
+        public HttpHeadersLogValue(Kind kind, HttpHeaders headers, HttpHeaders contentHeaders)
+        {
+            _kind = kind;
+
+            Headers = headers;
+            ContentHeaders = contentHeaders;
+        }
+
+        public HttpHeaders Headers { get; }
+
+        public HttpHeaders ContentHeaders { get; }
+        
+        private List<KeyValuePair<string, object>> Values
+        {
+            get
+            {
+                if (_values == null)
+                {
+                    var values = new List<KeyValuePair<string, object>>();
+                    
+                    foreach (var kvp in Headers)
+                    {
+                        values.Add(new KeyValuePair<string, object>(kvp.Key, kvp.Value));
+                    }
+
+                    if (ContentHeaders != null)
+                    {
+                        foreach (var kvp in ContentHeaders)
+                        {
+                            values.Add(new KeyValuePair<string, object>(kvp.Key, kvp.Value));
+                        }
+                    }
+
+                    _values = values;
+                }
+
+                return _values;
+            }
+        }
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (index < 0 || index >= Count)
+                {
+                    throw new IndexOutOfRangeException(nameof(index));
+                }
+
+                return Values[index];
+            }
+        }
+
+        public int Count => Values.Count;
+
+        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+        {
+            return Values.GetEnumerator();
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return Values.GetEnumerator();
+        }
+
+        public override string ToString()
+        {
+            return _kind.ToString();
+        }
```

Comment:
Overriding `ToString()` to do something less useful. SEQ likes to ToString() things and record that in addition to the key-value pairs.

Reply:
I think we moved to using ToString for most things instead of a Format method (https://github.com/aspnet/Hosting/blob/a997ceba98c40a98d2fd48190cba6d12abf105e7/src/Microsoft.AspNetCore.Hosting/Internal/HostingLoggerExtensions.cs#L137). 

Why not put this logic in ToString instead?

Reply:
Yeah it turns out this doesn't really make a difference